### PR TITLE
Jetpack Compose Support

### PR DESCRIPTION
**Pull request by abd3lraouf**
[//]: # (This is a comment line in markdown and will not be visible)
[//]: # (please remove unused lines)

### 🎯 Goal (required)

[//]: # (describe why we are making this change)
[//]: # (replace XYZ with the issue number)

Hello Jetpack Compose 🍀

### 🛠 Implementation details (required)

[//]: # (how do you implement this PR? Explain your solution)

- add new compose screen.
- add change user button
- add a toolbar with Storyteller SDK versions
- add Pull to refresh  behviour

### 🎨 UI Changes (optional)

[//]: # (explain UI changes)
[//]: # (Add relevant screenshots if it's UI related)
[//]: # (drag and drop images and replace their urls in the following html table)

<table>
  <tr>
    <td>1</td>
    <td>2</td>
    <td>3</td>
  </tr>
  <tr>
    <td></td>
    <td></td>
    <td></td>
  </tr>
 </table>

### 🧪 Testing (optional)

[//]: # (explain how to test your changes)
1. Set the Api key in the application class.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -27,23 +31,62 @@ android {
...
+
 
 }
 
 dependencies {
-    implementation fileTree(dir: 'libs', include: ['*.jar'])
-    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    /**
+     * Storyteller Compose SDK, it contains the latest core SDK transitively*/
+    def storyteller_compose_version = "0.0.1-local-main-e11076e"
```
Comment (fnabrdal) [inline app/build.gradle:55]: Well I'm now leaning towards an option with 2 dependencies SDK+Compose when we want to use compose. Just so our sample has SDK dependency as main one. Let's discuss with @DaveMiscampbell

File: app/src/main/java/com/example/storytellerSampleAndroid/SampleApp.kt [hunk 2]
```diff
@@ -24,7 +23,7 @@ class SampleApp : Application() {
                For more info, see - https://www.getstoryteller.com/documentation/android/getting-started#SDKInitialization
                 */
             Storyteller.initialize(
-                apiKey = "[APIKEY]",
+                apiKey = "   [APIKEY]",
```
Comment (fnabrdal) [inline app/src/main/java/com/example/storytellerSampleAndroid/SampleApp.kt:26]: ?

File: app/src/main/java/com/example/storytellerSampleAndroid/theme/Color.kt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.example.storytellerSampleAndroid.theme
+
+import androidx.compose.ui.graphics.Color
```
Comment (fnabrdal) [inline app/src/main/java/com/example/storytellerSampleAndroid/theme/Color.kt:3]: maybe we can put compose classes in the subpackage `com.example.storytellerSampleAndroid.compose`

File: app/src/main/java/com/example/storytellerSampleAndroid/SampleApp.kt (hunk 2 above)
Comment (abd3lraouf) [reply app/src/main/java/com/example/storytellerSampleAndroid/SampleApp.kt:26]: I was testing something and forgot to revert this one. :onit:

File: app/src/main/java/com/example/storytellerSampleAndroid/theme/Theme.kt [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+import androidx.compose.material3.dynamicLightColorScheme
+import androidx.compose.material3.lightColorScheme
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.SideEffect
+import androidx.compose.ui.graphics.toArgb
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalView
+import androidx.core.view.WindowCompat
+
+private val DarkColorScheme = darkColorScheme(
```
Comment (fnabrdal) [inline app/src/main/java/com/example/storytellerSampleAndroid/theme/Theme.kt:18]: We should probably agree on the colors we use. But they look fine for me.

File: app/src/main/java/com/example/storytellerSampleAndroid/theme/Type.kt [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+val Typography = Typography(
+  bodyLarge = TextStyle(
+    fontFamily = FontFamily.Default,
+    fontWeight = FontWeight.Normal,
+    fontSize = 16.sp,
+    lineHeight = 24.sp,
+    letterSpacing = 0.5.sp
+  )
+  /* Other default text styles to override
+    titleLarge = TextStyle(
```
Comment (fnabrdal) [inline app/src/main/java/com/example/storytellerSampleAndroid/theme/Type.kt:19]: Do we need them commented at all?

File: app/src/main/java/com/example/storytellerSampleAndroid/JetpackComposeViewModel.kt [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  private var _refreshing = mutableStateOf(false)
+  val refreshing: State<Boolean> get() = _refreshing
+
+  fun startRefreshing() {
+    _refreshing.value = true
+  }
+
+  fun stopRefreshing() {
+    viewModelScope.launch {
+      delay(2000)
```
Comment (fnabrdal) [inline app/src/main/java/com/example/storytellerSampleAndroid/JetpackComposeViewModel.kt:20]: why delay?

File: app/src/main/java/com/example/storytellerSampleAndroid/theme/Type.kt (hunk 5 above)
Comment (abd3lraouf) [reply app/src/main/java/com/example/storytellerSampleAndroid/theme/Type.kt:19]: it's a leftover from the compose sample.

File: app/src/main/java/com/example/storytellerSampleAndroid/theme/Theme.kt (hunk 4 above)
Comment (abd3lraouf) [reply app/src/main/java/com/example/storytellerSampleAndroid/theme/Theme.kt:18]: maybe @jasongilmour can provide assistance here ?

File: app/src/main/java/com/example/storytellerSampleAndroid/theme/Color.kt (hunk 3 above)
Comment (abd3lraouf) [reply app/src/main/java/com/example/storytellerSampleAndroid/theme/Color.kt:3]: Totally agree 🚧 :onit:

File: app/src/main/java/com/example/storytellerSampleAndroid/JetpackComposeActivity.kt [hunk 7]
```diff
@@ -0,0 +1,388 @@
...
+    }
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    Storyteller.storytellerDelegate = this
+    storytellerComposeController = StorytellerComposeController().bind(this)
+
+    setContent {
+      StorytellerSampleComposeTheme {
```
Comment (fnabrdal) [inline app/src/main/java/com/example/storytellerSampleAndroid/JetpackComposeActivity.kt:90]: I'm curious if we can move this whole Composable to another file?

Comment (abd3lraouf) [reply app/src/main/java/com/example/storytellerSampleAndroid/JetpackComposeActivity.kt:90]: Yes we can extract everything

**Review by DaveMiscampbell — COMMENTED** [review]
Need to make sure we update the README with details of the new sample and how to build it - you can look at how we set out the Web SDK sample and do the same thing - otherwise, looks good - good work!

**Pull request merged**